### PR TITLE
fix: metadata api exposed also inherited/linked tables from other schemas. closes #3022

### LINKED ISSUE #2982: Table explorer is broken for ontology tables
**What version of EMX2 are you using (see footer)**
v10.0.5-SNAPSHOT

**Describe the bug**
When opening an ontology table in the table explorer I get an error.

**To Reproduce**
1. Go to 'https://emx2.test.molgenis.org'
2. Click on 'catalogue-demo'
4. Click on tables
5. Click on 'Age Groups'

**Expected behavior**
The age groups ontology is shown in the table explorer.

**Actual result**
I see the following error in the browser:
```
Invalid syntax with offending token ‘}’ at line 10 column 11
```

And the web console log gives the following error:
```
[Error] Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating ‘t[this.tableId]’)
	(anonymous function) (index-2d010d34.js:151:27492)
```

**Additional context**
Probably related to the big id story?

### LINKED ISSUE #3022: Duplicate tables in schemas with references to another schema
**What version of EMX2 are you using (see footer)**
10.4.0

**Describe the bug**
Tables within a schema are duplicated. Seems to only be the case when the datamodel contains references to another schema (refSchema) and to tables that are not ontology tables.
image

**To Reproduce**
1. Go to https://emx2.dev.molgenis.org/test_staging/tables/#/
2. See that some, but not all, tables are duplicated
3. This is also the case in the up/download section: https://emx2.dev.molgenis.org/test_staging/updownload/
4. Only happens when there are references to non ontology tables in another schema.

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by mswertz**
closes #3022 
closes #2982

### CODE REVIEW

File: backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/json/Schema.java [hunk 1]
```diff
@@ -25,7 +25,7 @@ public Schema(SchemaMetadata schema, boolean minimal) {
             .map(entry -> new Setting(entry.getKey(), entry.getValue()))
             .toList();
     List<TableMetadata> list = new ArrayList<>();
-    list.addAll(schema.getTablesIncludingExternal());
```
Comment (connoratrug) [inline backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/json/Schema.java:28]: why would getTablesIncludingExternal fetch non related tables ?

**Comment by davidruvolo51** [comment]
This resolves the duplicate table issue. However, I noticed that if a table is extended, the table type throws an error. Schemas can be imported only if the table type is removed

In emx2.dev.molgeniscloud.org:

- create a new schema called "mainDB", and import the parent.csv file (mainDB is hardcoded in the child schema)
- create a second schema and import the child.csv file

[parent.csv](https://github.com/molgenis/molgenis-emx2/files/13447543/parent.csv)
[child2.csv](https://github.com/molgenis/molgenis-emx2/files/13447544/child2.csv)

File: backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/json/Schema.java (hunk 1 above)
Comment (connoratrug) [reply backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/json/Schema.java:28]: getTablesIncludingExternal is no longer called after this change and should be removed, sonar should have found this ( maybe it did ? )

**Review by connoratrug — CHANGES_REQUESTED** [review]

File: backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/json/Schema.java (hunk 1 above)
Comment (mswertz) [reply backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/json/Schema.java:28]: will double check and delete if not used

**Comment by mswertz** [comment]
And I should add a test to prevent regressions

**Comment by davidruvolo51** [comment]
This is probably a separate issue. Regarding table inheritance, an error should be thrown when a user would like to delete a schema that is inherited. Currently, the base schema is automatically deleted without warning. When an inherited schema is deleted, you will receive an error that the base schema does not exist.

**Comment by mswertz** [comment]
> This resolves the duplicate table issue. However, I noticed that if a table is extended, the table type throws an error. Schemas can be imported only if the table type is removed
> 
> In emx2.dev.molgeniscloud.org:
> 
> * create a new schema called "mainDB", and import the parent.csv file (mainDB is hardcoded in the child schema)
> * create a second schema and import the child.csv file
> 
> [parent.csv](https://github.com/molgenis/molgenis-emx2/files/13447543/parent.csv) [child2.csv](https://github.com/molgenis/molgenis-emx2/files/13447544/child2.csv)

is this a new error?

**Comment by mswertz** [comment]
> This is probably a separate issue. Regarding table inheritance, an error should be thrown when a user would like to delete a schema that is inherited. Currently, the base schema is automatically deleted without warning. When an inherited schema is deleted, you will receive an error that the base schema does not exist.

I though foreign key constraints would prevent this. I added issue.

**Pull request merged**